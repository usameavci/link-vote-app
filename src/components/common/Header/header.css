.common-header {
  display: flex;
  padding: 1rem 0;
  font-size: 2rem;
  align-items: center;
  border-bottom: 1px solid #000;
  justify-content: space-between;
}
