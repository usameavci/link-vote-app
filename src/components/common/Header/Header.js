import React from "react";

import CommonLogo from "../Logo/Logo";
import CommonHeaderText from "../HeaderText/HeaderText";

import "./header.css";

const Header = () => {
  return (
    <div data-testid="common-header" className="common-header">
      <CommonLogo />
      <CommonHeaderText />
    </div>
  );
};

export default Header;
