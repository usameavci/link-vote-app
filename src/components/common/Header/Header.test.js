import React from "react";
import renderer from "react-test-renderer";
import { BrowserRouter as Router } from "react-router-dom";
import { render } from "@testing-library/react";

import Header from "./Header";

describe("Header component tests", () => {
  it("renders without crashing", async () => {
    const component = (
      <Router>
        <Header />
      </Router>
    );

    const { getByTestId } = render(component);

    expect(getByTestId("common-header")).toBeInTheDocument();
  });

  it("matches snapshots", () => {
    const component = (
      <Router>
        <Header />
      </Router>
    );

    const tree = renderer.create(component);

    expect(tree.toJSON()).toMatchSnapshot();
  });
});
