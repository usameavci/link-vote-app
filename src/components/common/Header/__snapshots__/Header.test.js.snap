// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header component tests matches snapshots 1`] = `
<div
  className="common-header"
  data-testid="common-header"
>
  <a
    data-testid="common-logo"
    href="/"
    onClick={[Function]}
  >
    <img
      alt="Hepsiburada"
      className="common-logo"
      src="logo.jpg"
    />
  </a>
  <div
    className="common-header-text"
    data-testid="common-header-text"
  >
    <span>
      Link
    </span>
    <span>
      VOTE
    </span>
    <span>
      Challenge
    </span>
  </div>
</div>
`;
